packages/launcher/src/components/launch/Launch.tsx

```diff
@@ -73,35 +79,118 @@ export default function Launch(): ReactElement {
       setAxonVersions(args);
       setAxonVersion(args[0]);
     });
     ipcRenderer.send("request-tags");
     ipcRenderer.send("request-version");
   };
+  const getTagsFromSystem = async () => {
+    const connected = await docker.isConnected();
+    if (connected) {
+      const images = await docker.getImages();
+      if (images !== null && images.length > 0) {
+        const imageMap = new Map();
+        for (let i = 0; i < images.length; i++) {
+          const tmpTag = images[i].RepoTags[0].split(":")[1];
+          if (imageMap.has(tmpTag)) {
+            imageMap.set(tmpTag, imageMap.get(tmpTag) + 1);
+          } else {
+            imageMap.set(tmpTag, 1);
+          }
+        }
+        const tmpTags: string[] = [];
+        imageMap.forEach((value: number, key: string) => {
+          if (value >= 6) tmpTags.push(key);
+        });
+        console.log(tmpTags);
+        setAxonVersions(tmpTags);
+        setAxonVersion(tmpTags[0]);
+      }
+    }
+  };
+  const getInternetConnection = async () => {
+    const ipcRenderer: IpcRenderer = window.require("electron").ipcRenderer;
+    await ipcRenderer.on("internet-status", (event, arg: boolean) => {
+      setInternetConnection(arg);
+      setCheckedInternetConnection(true);
+      if (!arg) {
+        // need to look for own tags
+        getTagsFromSystem();
+      }
+    });
+    ipcRenderer.send("request-internet");
+  };
+
   if (!requested) {
     getVersions();
     setRequested(true);
     return <CircularProgress />;
   }
 
+  if (!checkedInternetConnection) {
+    getInternetConnection();
+    return <CircularProgress className={classes.progress} />;
+  }
+
   const handleClose = () => {
     setClicked(true);
   };
 
+  const handleImageDialogClose = () => {
+    setClickedImageDialog(true);
+  };
+
   const startContainer = async () => {
     const connected = await docker.isConnected();
     if (connected) {
-      // setPulling(true);
-      setStatus("Pulling Axon image");
-      await docker.pullImage(axonVersion);
-      // setPulling(false);
-      setStatus("Finished pulling.");
+      if (internetConnection) {
+        console.log("Valid Internet Connection");
+        setStatus("Pulling Axon image");
+        await docker.pullImage(axonVersion);
+        // setPulling(false);
+        setStatus("Finished pulling.");
+        setHasImages(true);
+      } else {
+        console.log("No Internet Connection Detected, Skipping Pulling Images");
+        const images = await docker.getImages();
+        if (images !== null && images.length > 0) {
+          let count = 0;
+          for (let i = 0; i < images.length; i++) {
+            // Check if axonVersion is equal to the tag on the image name
+            if (axonVersion === images[i].RepoTags[0].substring(images[i].RepoTags[0].length - axonVersion.length)) {
+              count++;
+              console.log(images[i].RepoTags[0]);
+            }
+          }
```

Comment:
Use [every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every) or [for ... of](https://www.typescriptlang.org/docs/handbook/iterators-and-generators.html#forof-statements) to avoid manual iteration